src/Decompiler/Analysis/SsaTransform.cs

```diff
@@ -1508,6 +1502,28 @@ private void ReplaceBy(SsaIdentifier sidOld, SsaIdentifier idNew)
                     }
                 }
             }
+
+            private void ReplaceStackDefs(
+                SsaBlockState bs,
+                SsaIdentifier sidOld,
+                SsaIdentifier sidNew)
+            {
+                if (!(sidOld.Identifier.Storage is StackStorage stack))
+                    return;
+                var offsetInterval = Interval.Create(
+                    stack.StackOffset,
+                    stack.StackOffset + sidOld.Identifier.DataType.Size);
+                var ints = bs.currentStackDef
+                    .GetIntervalsOverlappingWith(offsetInterval)
+                    .ToArray();
+                foreach (var de in ints)
```

Comment:
If this is performance critical, perhaps we shouldn't materialize the `ints` into an array, but feed the intervals directly into the following `foreach` loop.

Reply:
Done. See 4eba33e